File: graphql-test/src/main/java/com/junyss/graphqltest/common/util/GraphQLSupportUtil.java
package com.junyss.graphqltest.common.util;

import java.util.Collections;
import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.stereotype.Component;

import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Component
public class GraphQLSupportUtil {

	public static <T> List<T> pageToList(Page<T> pageTypeDto) {
		if (pageTypeDto.getTotalElements() == 0 && !pageTypeDto.hasContent()) {
			return Collections.emptyList();
		}

		return pageTypeDto.getContent();
	}
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/common/util/GraphQLSupportUtil.kt
package com.junyharangstudy.kotlingraphqltest.common.util

import org.springframework.data.domain.Page
import org.springframework.stereotype.Component

@Component
class GraphQLSupportUtil {
    companion object {
        fun <T> pageToList(pageTypeDto: Page<T>): List<T> {
            if (pageTypeDto.totalElements == 0L && !pageTypeDto.hasContent()) {
                return emptyList()
            }

            return pageTypeDto.content
        }
    }
}